File: leetcode/src/main/java/com/thanos/algorithm/jav/TwoSum.java
package com.thanos.algorithm.jav;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by solarknight on 2017/4/4.
 */
public class TwoSum {

  public int[] twoSum(int[] nums, int target) {
    Map<Integer, Integer> map = new HashMap<Integer, Integer>();

    for (int i = 0; i < nums.length; i++) {
      int tmp = target - nums[i];
      if (map.containsKey(tmp)) {
        return new int[]{map.get(tmp), i};
      } else {
        map.put(nums[i], i);
      }
    }
    return null;
  }
}

File: leetcode/src/main/java/com/thanos/algorithm/golang/two_sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for idx, v := range nums {
        if i, ok := m[target - v]; ok {
            return []int{i, idx}
        } else {
            m[v] = idx
        }
    }
    return nil
}